### PR TITLE
Milestone 10: Portfolio Preserve data

### PR DESCRIPTION
**Pull request by michael-duke**
## Portfolio milestone 10 : Preserve data

In this milestone @peterboro and @michael-duke :
- Used local storage to save user input, reset it as well.
- Have used @michael-duke's portfolio
- Followed JavaScript best practices to destructure objects.
- Implemented the Reset additional requirement.

### CODE REVIEW

**Comment by michael-duke** [comment]
#### Dear, reviewer

Happy reviewing 😄 from both @peterboro and me.

File: js/storage.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  const savedForm = JSON.parse(localStorage.getItem('form'));
+  if (savedForm) {
+    const { name, email, message } = form.elements;
+    name.value = savedForm.name;
+    email.value = savedForm.email;
+    message.value = savedForm.message;
+  }
+};
+
+export { onUpdate, refreshForm, resetForm };
```
Comment (zieeco) [inline js/storage.js:39]: - Please, kindly ensure that when the user refreshes the page after filling your form without submitting it, all inputs fields should be populated as of now, the textarea doesn't get populated, only the name and email inputs get populated.

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @gnuion,


Great job so far! 👏 
You've done well, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- No linter errors.✔️
- Nice design. ✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._





## Optional suggestions
- N/A

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by michael-duke** [comment]
@zieeco thank you for your review. Have you clicked outside the textarea and then refresh, if you don't then the message won't be saved

**Comment by michael-duke** [comment]
I kindly suggest you press tab after each entry, and then refresh the browser, thank you 😄

**Comment by michael-duke** [comment]
**Dear, reviewer**
Kindly note previous comments I made, regarding pressing tab after entering the inputs and textarea,

Happy coding 😄

**Review by ShahierNashaat — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Approved :ok_hand: :ok_hand: :ok_hand:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by michael-duke** [comment]
@ShahierNashaat  thank you for you approval on behalf of the team

**Pull request merged**